README.md

```diff
@@ -11,12 +11,105 @@
 to build a number of internal developer tools, along with
 [cli-ui](https://github.com/shopify/cli-ui).
 
-## Example Usage
+## Getting Started
 
+To begin creating your first `cli-kit` application, run:
 ```bash
 gem install cli-kit
 cli-kit new myproject
 ```
 
-You can see example usage [here](https://github.com/Shopify/cli-kit-example). This app is similar to
-the one that the generator generates.
+Where `myproject` is the name of the application you wish to create.  Then, you will be prompted to
+select how the project consumes `cli-kit` and `cli-ui`.  The available options are:
+- Vendor (faster execution, more difficult to update dependencies)
+- Bundler (slower execution, easier dependency management)
+
+You're now ready to write your very first `cli-kit` application!
+
+## How do `cli-kit` Applications Work?
+
+The executable for your `cli-kit` app is stored in the "exe" directory.  To execute the app, simply
+run:
+```bash
+./exe/myproject
+```
+
+### Folder Structure
+* `/exe/` - Location of the executables for your application.
+* `/lib/` - Location of the resources for your app (modules, classes, helpers, etc).
+    * `myproject.rb` - The main file for your application.
+    * `myproject/` - Stores the various commands/entry points.
+        * `entry_point.rb` - This is the file that is first called from the executable. It handles 
+        loading and commands.
+        * `commands.rb` - Registers the various commands that your application is able to handle.
+        * `commands/` - Stores Ruby files for each command (help, new, add, etc).
+
+## Adding a New Command to your App
+
+### Registering the Command
+
+Let's say that you'd like your program to be able to handle a specific task, and you'd like to
+_register_ a new handler for the command for that task, like `myproject add` to add 2 numbers, like
+in a calculator app.
+To do this, open `/lib/myproject/commands.rb`. Then, add a new line into the module, like this:
+```ruby
+register :Add, 'add', 'myproject/commands/add'
+```
+
+The format for this is `register :<CommandClass>, '<command-at-cli>', '<path/to/command.rb>'`
+
+### Creating the Command Action
+
+The action for a specific command is stored in its own Ruby file, in the `/lib/myproject/commands/`
+directory.  Here is an example of the `add` command in our previous to-do app example:
+```ruby
+require 'myproject'
+
+module Myproject
+  module Commands
+    class Add < Myproject::Command
+      def call(args, _name)
+        # command action goes here
+      end
+
+      def self.help
+        # help or instructions go here
+      end
+    end
+  end
+end
+
+```
+
+The `call(args, _name)` method is what actually runs when the `myproject add` command is executed.
+
+- **Note:** The `args` parameter represents all the arguments the user has specified.
+
+Let's say that you are trying to compute the sum of 2 numbers that the user has specified as
+arguments.  For example:
+```ruby
+def call(_args, _name)
```

Comment:
Drop the `_` on `args`.  The `_` is a way of saying "this is my method signature (usually to conform to a duck type), but for this particular method I don't actually need this value".  You need `args`, but you don't need `_name` here.

Reply:
Cool!  I'll remove the `_`.  So I should remove the `_name` entirely?